### PR TITLE
PH/MIN-361 Parachain and Standalone runtimes

### PR DESCRIPTION
**Pull request by msifd**
**Description:**

Split runtime into two almost identical implementations: Parachain and Standalone.
Comparing to Parachain runtime, Standalone has additional grandpa pallet, benchmarks and tests.

In cli and service modules was added logic to start node in parachain mode and in standalone mode.
In chain_spec was split into two parts: "staging" parachain spec and dev/local standalone spec.

**Changes in storage or API:**

CLI changes:
- Added relaychain args
- New build-in chain-specs:
    - default - Local Testnet parachain
    - `dev` or `standalone-dev` - Standalone Dev
    - `standalone-local` - Standalone Local Testnet

**Checklist**

- [ ] Code is covered with tests
- [ ] Code is covered with benchmarks
- [ ] Fresh benchmark results included

### CODE REVIEW

**Comment by Apollo50** [comment]
Please, fix all warnings in compile time.

File: cli/Cargo.toml [hunk 1]
```diff
@@ -6,19 +6,40 @@ edition = "2018"
...
 [dependencies]
 structopt = "0.3.8"
+codec = { package = "parity-scale-codec", version = "2.0.0" }
+log = { version = "0.4.14", default-features = false }
+
 sp-core = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.8", optional = true }
 sc-cli = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.8", optional = true }
 sc-service = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.8", optional = true }
+sc-tracing = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.8" }
+sc-telemetry = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.8" }
```
Comment (kurotych) [inline cli/Cargo.toml:16]: Do we need `default-features = false` for there pallets?

File: rpc/Cargo.toml [hunk 2]
```diff
@@ -18,7 +18,7 @@ sc-rpc = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v
 orml-oracle-rpc = { git = "https://github.com/open-web3-stack/open-runtime-module-library" }
 
 # local dependencies
-node-minterest-runtime = { path = "../runtime", version = "0.6.0" }
+parachain-runtime = { path = "../runtime/parachain", version = "0.6.0" }
```
Comment (kurotych) [inline rpc/Cargo.toml]: Maybe we should use `minterest-parachain-runtime` naming?

File: runtime/parachain/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,955 @@
...
+}
+
+impl pallet_aura::Config for Runtime {
+	type AuthorityId = AuraId;
+}
+
+const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_SECOND * 2;
+
+parameter_types! {
+	// pub const ReservedXcmpWeight: Weight = MAXIMUM_BLOCK_WEIGHT / 4;
```
Comment (kurotych) [inline runtime/parachain/src/lib.rs]: Do we need this commented line?

File: runtime/standalone/Cargo.toml [hunk 4]
```diff
@@ -2,27 +2,19 @@
 authors = ["Minterest <https://github.com/minterest-defi>"]
 edition = "2018"
 license = "Unlicense"
-name = "node-minterest-runtime"
+name = "standalone-runtime"
```
Comment (kurotych) [inline runtime/standalone/Cargo.toml]: I suggest `minterest-standalone-runtime`

File: runtime/standalone/src/weights/controller.rs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+// ./target/release/minterest
+// benchmark
+// --chain=dev
+// --steps=50
+// --repeat=20
+// --pallet=controller
+// --extrinsic=*
+// --execution=wasm
+// --wasm-execution=compiled
+// --output=./runtime/src/weights/controller.rs
```
Comment (kurotych) [inline runtime/standalone/src/weights/controller.rs]: Could you fix this? 
`--output=./runtime/src/weights/controller.rs` but file has different path `runtime/standalone/src/weights/controller.rs`
Just recalculate weights. 

The same below

File: scripts/export_para.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+set -euo pipefail
+IFS=$'\n\t'
+
+
+NODE=../target/release/minterest
+
+$NODE export-genesis-state --parachain-id 2000 > minterest-para-genesis-2000
+$NODE export-genesis-wasm > minterest-para-wasm
```
Comment (kurotych) [inline scripts/export_para.sh:9]: Add new line. The same below

File: scripts/export_para.sh [hunk 7]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+set -euo pipefail
+IFS=$'\n\t'
```
Comment (kurotych) [inline scripts/export_para.sh:3]: Is this unused variable?

**Review by kurotych — COMMENTED** [review]
Could you write an instruction how to use your scripts?

File: scripts/export_para.sh (hunk 7 above)
Comment (msifd) [reply scripts/export_para.sh:3]: no, its for strict bash syntax

File: cli/Cargo.toml (hunk 1 above)
Comment (msifd) [reply cli/Cargo.toml:16]: Nope

**Review by kurotych — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]

**Pull request merged**